src/tests/ftest/util/command_utils.py

```diff
@@ -467,25 +479,64 @@ def set_sub_command(self, value):
         self.sub_command.value = value
         self.get_sub_command_class()
 
-    def _get_result(self):
-        """Get the result from running the configured command.
-
-        Returns:
-            CmdResult: an avocado CmdResult object containing the command
-                information, e.g. exit status, stdout, stderr, etc.
+    def run(self):
+        """Run the command and assign the 'result' attribute.
 
         Raises:
-            CommandFailure: if the command fails.
+            CommandFailure: if there is an error running the command and the
+                CommandWithSubCommand.exit_status_exception attribute is set to
+                True.
+
+        Returns:
+            CmdResult: a CmdResult object containing the results of the command
+                execution.
 
         """
-        result = None
         try:
-            result = self.run()
+            self.result = super(CommandWithSubCommand, self).run()
         except CommandFailure as error:
             raise CommandFailure(
                 "<{}> command failed: {}".format(self.command, error))
+        return self.result
+
+    def _get_result(self, sub_command_list=None, **kwargs):
+        """Get the result from running the command with the defined arguments.
+
+        The optional sub_command_list and kwargs are used to define the command
+        that will be executed.  If they are excluded, the commnad will be run as
+        it currently defined.
```

Comment:
```suggestion
        that will be executed.  If they are excluded, the command will be run as defined.
```
or something similar.

Reply:
Thanks, I like that better.  I'll apply the change in a follow-on PR.